### PR TITLE
[ONNX] Extend ONNX Frontend with BitwiseOr-18 operator

### LINKED ISSUE #20791: [Good First Issue]: Extend ONNX Frontend with BitwiseOr-18 operator
### Context

Neural networks are graphs consisting of nodes called operators. Each operator corresponds to a mathematical function, usually described in framework's documentation or an AI standard, such as ONNX.
OpenVINO ONNX Frontend is a component responsible for working with ONNX graphs and requires implementation of different ONNX operators in order to use ONNX models.

This task requires extending OpenVINO ONNX Frontend with `BitwiseOr` operator, using the existing [ov::op::v13::BitwiseOr](https://github.com/openvinotoolkit/openvino/blob/master/src/core/include/openvino/op/bitwise_or.hpp).

Necessary help will be provided by ONNX Fronted team.

### What needs to be done?

Operator details can be found in [ONNX Operators](https://github.com/onnx/onnx/blob/main/docs/Operators.md#BitwiseOr)
More details can be found in [ONNX Changelog](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#BitwiseOr-18)
1. Create `.hpp` and `.cpp` files [here](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/onnx/frontend/src/op)
2. Prepare an implementation of this operator in form of a function, `bitwise_or` should be placed in [opset 1 namespace](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/op/unique.hpp#L13).
3. Register the function in [ops_bridge.cpp](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/ops_bridge.cpp#L554-L560) while keeping alphabetical order
4. Create test model(s) in [ONNX models directory](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/onnx/tests/models/). OpenVINO test infrastructure then converts prototxt files to ONNX models - you will use those models later in tests
5. Add tests covering all use cases [here](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/tests/onnx_import.in.cpp)
6. Check [Python xfailed tests](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/tests/tests_python/test_backend.py) to find a test marked as a xfailed for added functionality. If any exist - remove corresponding lines and try to verify by using cmdline "python -m pytest -k name_of_test".
More details in [adding operators to ONNX Frontend guide](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/how_to_add_op.md)

### Example Pull Requests

- https://github.com/openvinotoolkit/openvino/pull/15076
- https://github.com/openvinotoolkit/openvino/pull/14936
- https://github.com/openvinotoolkit/openvino/pull/12510
- https://github.com/openvinotoolkit/openvino/pull/19428

### Resources

- [Contribution guide - start here!](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- Guides on [ONNX Frontend architecture](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/architecture.md) and [ONNX Frontend tests](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/tests.md) are also available

### Contact points
@mitruska
@gkrivor
@p-wysocki 

### Ticket
118977

### PR DESCRIPTION
**Pull request by rghvsh**
### Details:
 - Implemented ONNX frontend BitwiseOr operator and corresponding tests.

### Tickets:
 - Closes #20791

### CODE REVIEW

**Comment by mlukasze** [comment]
build_jenkins

**Comment by rghvsh** [comment]
Hello @mlukasze @gkrivor !
Under the clang-format check log it shows :
>Raw Output:
>src/frontends/onnx/frontend/src/op/bitwise_or.cpp:5:-#include "op/bitwise_or.hpp"  
>src/frontends/onnx/frontend/src/op/bitwise_or.cpp:5:+#include "op/bitwise_or.hpp"
>Error: Process completed with exit code 1.

I ran clang-format command on the code there was no change, what does this error refer to?

Thanks
Raghav

File: src/frontends/onnx/frontend/src/op/bitwise_or.cpp [hunk 1]
```diff
@@ -0,0 +1,24 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "op/bitwise_or.hpp"
```
Comment (mmikolajcz) [inline src/frontends/onnx/frontend/src/op/bitwise_or.cpp:5]: Please fix code style. You can follow generated suggestions from `Files changed` github tab, or use either `cmake --build ./build --target clang_format_fix_all` to automatically fix formatting or `cmake --build ./build --target clang_format_check_all` to check it locally.

**Review by mmikolajcz — COMMENTED** [review]
Thank you for your contribution, PR looks great!
I've left one comment regarding code style with some guides how to fix it.

**Comment by mmikolajcz** [comment]
> Hello @mlukasze @gkrivor ! Under the clang-format check log it shows :
> 
> > Raw Output:
> > src/frontends/onnx/frontend/src/op/bitwise_or.cpp:5:-#include "op/bitwise_or.hpp"
> > src/frontends/onnx/frontend/src/op/bitwise_or.cpp:5:+#include "op/bitwise_or.hpp"
> > Error: Process completed with exit code 1.
> 
> I ran clang-format command on the code there was no change, what does this error refer to?
> 
> Thanks Raghav

I think issue is due to 2 whitespaces at the end of line `+#include "op/bitwise_or.hpp"  `. Probably you would need to turn on whitespace rendering option in your IDE to catch it

**Review by mmikolajcz — APPROVED** [review]
LGTM, thanks for quick fix, lets wait for CI to pass :)

**Comment by mmikolajcz** [comment]
build_jenkins

**Comment by gkrivor** [comment]
Hi, wheels failed due to issue with uint64 test case OnnxBackendNodeModelTest.test_bitwise_or_ui64_bcast_3v1d_cpu

Please, add it to skipped list as for bitwise_and.

**Comment by mmikolajcz** [comment]
build_jenkins

**Comment by rghvsh** [comment]
Hi! @mmikolajcz @gkrivor Is there a community forum like a discord/slack channel or a mailing list for OpenVINO?

Thanks
Raghav

**Comment by mlukasze** [comment]
hey @rghvsh 

discord will be available after new year, we are preparing for xmas break and we might not be able to serve communication with you guys in a proper way.

**Pull request merged**